### PR TITLE
executor: privilege check USE command

### PR DESCRIPTION
**Pull request by morgo**
### What problem does this PR solve? 

In MySQL, a `use DB` will return a privilege error for users who would not have access to the DB irrespective on whether it exists.

This PR changes TiDB to behave similarly and not leak that a schema exists.

### What is changed and how it works?

The `use DB` (and protocol commands that are similar) now may return an error.

### Check List 

Tests 

 - Unit test
 - Manual test (add detailed scripts or steps below)

Code changes

 - minimal

Side effects

 - Breaking backward compatibility for broken clients

Related changes

 - Need to be included in the release note

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tidb/8418)

### CODE REVIEW

File: server/server_test.go [hunk 1]
```diff
@@ -619,34 +619,36 @@ func runTestShowProcessList(c *C) {
...
 		dbt.mustExec(`CREATE USER 'authtest2'@'localhost' IDENTIFIED BY '123';`)
+		dbt.mustExec(`GRANT ALL on test.* to 'authtest2'@'localhost'`)
 		dbt.mustExec(`FLUSH PRIVILEGES;`)
 	})
 	runTests(c, func(config *mysql.Config) {
 		config.User = "authtest2"
 		config.Passwd = "123"
 	}, func(dbt *DBTest) {
-		dbt.mustExec(`USE mysql;`)
+		dbt.mustExec(`USE information_schema;`)
```
Comment (winkyao) [inline server/server_test.go]: Add some deny access cases?

Comment (morgo) [reply server/server_test.go]: Sure.  I've added it in privilege testing.

**Comment by zz-jason** [comment]
@tiancaiamao @lysu @winkyao PTAL

**Comment by tiancaiamao** [comment]
LGTM

**Pull request merged**